### ISSUE TITLE
if method == a_method makes the concept of passing methods

### ISSUE THREAD

**Issue by haesleinhuepf** [BiAPoL/biapol-utilities#11]
We should reformulate code containing
```
    if method == intersection_over_union_matrix:
        threshold = kwargs.get('iou_threshold', 0.25)
```
because we pass the method to prevent code like `if method==`. The ultimate goal of passing methods is that the code works with whatever method we pass no matter how this method works.

Let's think about a second implementation of such a method, e.g. generating a [centroid-distance-matrix](https://github.com/clEsperanto/pyclesperanto_prototype/blob/master/demo/neighbors/generate_centroid_distance_matrix.ipynb) and then connecting labels that are within a certain distance range. 

It should at the end work as easy as demonstrated in [this notebook](https://github.com/BiAPoL/Bio-image_Analysis_with_Python/blob/main/python_basics/12_functional_parameters.ipynb).

**Comment by haesleinhuepf** [comment]
I think, we want to pass a function with this signature:
```
intersection_over_union_matrix_below_threshold(label_image1, label_image2, minimum_overlap_threshold)
```

**Comment by jo-mueller** [comment]
Fixed in PR #21 -closing this now :)
